contracts/staking/fuse/StakingRewards.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StakingRewards is ReentrancyGuard, Pausable, Ownable {
+    using SafeERC20 for IERC20;
+
+    /* ========== STATE VARIABLES ========== */
+
+    IERC20 public rewardsToken;
+    IERC20 public stakingToken;
+    uint256 public periodFinish = 0;
+    uint256 public rewardRate = 0;
+    uint256 public rewardsDuration = 7 days;
+    uint256 public lastUpdateTime;
+
+    uint256[] public rewardPerTokenStoredAt;
+
+    mapping(address => uint256) public userRewardPerTokenPaid;
+    mapping(address => uint256) public rewards;
+
+    uint256 private _totalSupply;
+    mapping(address => uint256) private _pendingStakes;
```

Comment:
pendingStakes is a single global variable.
a staker has the following fields in his struct
rewards - paid so far
rewardPerTokenPaidIndex - the index to rewardPerTokenStoredAt where user was last paid
avgGivebackRatio
stake/balance - how much user has staked

When a staker stakes, his total stake is increased and so are the pendingStakes
once rewardPerToken() is called pedningStakes are added to totalSupply and are reset to 0.
dont forget we dont pay rewards per block here so basically it should be rewardPerToken(uint256 tokenRewards)

Reply:
The code was removed.

Reply:
not it wasnt see my previous comment plus you are using the term "pendingStakes" incorrectly

Reply:
Done